src/python/grpcio/grpc/framework/crust/_control.py

```diff
@@ -0,0 +1,545 @@
+  """Values indicating whether and how an RPC has terminated.
+
+  Attributes:
+    terminated: A boolean indicating whether or not the RPC has terminated.
+    abortion: A face.Abortion value describing the RPC's abortion or None if the
+      RPC did not abort.
+    abortion_error: A face.AbortionError describing the RPC's abortion or None
+      if the RPC did not abort.
+  """
+
+_NOT_TERMINATED = _Termination(False, None, None)
+
+_OPERATION_OUTCOME_TO_TERMINATION_CONSTRUCTOR = {
+    base.Outcome.COMPLETED: lambda *unused_args: _Termination(True, None, None),
+    base.Outcome.CANCELLED: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.CANCELLED, *args),
+        face.CancellationError(*args)),
+    base.Outcome.EXPIRED: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.EXPIRED, *args),
+        face.ExpirationError(*args)),
+    base.Outcome.LOCAL_SHUTDOWN: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.LOCAL_SHUTDOWN, *args),
+        face.LocalShutdownError(*args)),
+    base.Outcome.REMOTE_SHUTDOWN: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.REMOTE_SHUTDOWN, *args),
+        face.RemoteShutdownError(*args)),
+    base.Outcome.RECEPTION_FAILURE: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.NETWORK_FAILURE, *args),
+        face.NetworkError(*args)),
+    base.Outcome.TRANSMISSION_FAILURE: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.NETWORK_FAILURE, *args),
+        face.NetworkError(*args)),
+    base.Outcome.LOCAL_FAILURE: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.LOCAL_FAILURE, *args),
+        face.LocalError(*args)),
+    base.Outcome.REMOTE_FAILURE: lambda *args: _Termination(
+        True, face.Abortion(face.Abortion.Kind.REMOTE_FAILURE, *args),
+        face.RemoteError(*args)),
+}
+
+
+def _wait_once_until(condition, until):
+  if until is None:
+    condition.wait()
+  else:
+    remaining = until - time.time()
+    if remaining < 0:
+      raise future.TimeoutError()
+    else:
+      condition.wait(timeout=remaining)
+
+
+def _done_callback_as_operation_termination_callback(
+    done_callback, rendezvous):
+  def operation_termination_callback(operation_outcome):
+    rendezvous.set_outcome(operation_outcome)
+    done_callback(rendezvous)
+  return operation_termination_callback
+
+
+def _abortion_callback_as_operation_termination_callback(
+    rpc_abortion_callback, rendezvous_set_outcome):
+  def operation_termination_callback(operation_outcome):
+    termination = rendezvous_set_outcome(operation_outcome)
+    if termination.abortion is not None:
+      rpc_abortion_callback(termination.abortion)
+  return operation_termination_callback
+
+
+class Rendezvous(base.Operator, future.Future, stream.Consumer, face.Call):
+  """A rendez-vous for the threads of an operation.
```

Comment:
Apparently that's the spelling that I used when I wrote the comment... one year ago. I'm not attached to it and can drop the dash if you think it helps the reader.